### ISSUE TITLE
option to include a xsl reference in xml output

### ISSUE THREAD

**Issue by pvgenuchten** [geopython/pycsw#888]
At https://files.isric.org/projects/geopearl/250m/sl1_som3035_arable_250m.xml we applied an old xml convention to add a xsl reference to an xml so it is loaded in browsers as human readable

![image](https://github.com/geopython/pycsw/assets/299829/95d29603-07ca-48b0-8ce5-97b1086c7b23)

Colleagues asked if this convention could also be applied on xml output from pycsw, so when they download the xml, and open the xml locally, it would be displayed human readable.

For this to function we'd need to:

- publish an xsl at a public location
- in xml output (single record only?) add a reference to that xsl
- verify if this is standards compliant?

**Comment by tomkralidis** [comment]
@pvgenuchten I guess this would be different than our [existing XSLT support](https://docs.pycsw.org/en/latest/xslt.html) (which returns the output of the transformation).  We could add a `server.xsl` block [in `pycsw/core/server:Csw._write_response`](https://github.com/geopython/pycsw/blob/master/pycsw/server.py#L669) that would inject a given XSL at the root of the output document.  Keep in mind this would apply to single record XML output in any `outputSchema`.  Would this work for you?

I think would be compliant as the XSL would be ignored by client machinery, but triggered by those on a browser, say.

**Comment by pvgenuchten** [comment]
correct, this would only apply to single record discplay (without the csw-response wrapper)
would be great to have such a configuration option somewhere
